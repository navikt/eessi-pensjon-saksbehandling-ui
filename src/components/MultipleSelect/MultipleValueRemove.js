import React from 'react'
import Icons from 'components/Icons'
import { components } from 'react-select'
const MultipleValueRemove = (props) => {
  return <components.MultiValueRemove {...props} className='c-multipleSelect-multipleValueRemove'>
    <Icons kind='close' />
  </components.MultiValueRemove>
}

export default MultipleValueRemove
