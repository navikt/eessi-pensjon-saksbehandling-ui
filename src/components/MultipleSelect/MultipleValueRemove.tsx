import React, { useState } from 'react'
import FilledRemoveCircle from 'assets/icons/filled-version-remove-circle'
import styled from 'styled-components'

const FlexDiv = styled.div`
  display: flex;
  cursor: pointer;
  &:hover {
    background-color: transparent !important;
  }
`
const MultipleValueRemove = (props: any) => {
  const [hover, setHover] = useState<boolean>(false)
  const { selectProps, innerProps } = props
  const theme = selectProps.selectProps.theme

  const visibility = selectProps.isDisabled ? 'hidden' : 'visible'
  return (
    <FlexDiv
      {...innerProps}
      style={{}}
      data-test-id='c-multipleselect-multiplevalueremove'
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}

    >
      <FilledRemoveCircle
        style={{ visibility: visibility }}
        color={
        theme[hover ? 'main-interactive-color' : 'main-font-color']
      }
      />
    </FlexDiv>

  )
}
export default MultipleValueRemove
