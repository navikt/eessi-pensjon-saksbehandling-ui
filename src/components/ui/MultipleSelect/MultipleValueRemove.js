import React from 'react'
import Icons from 'components/ui/Icons'

const MultipleValueRemove = (props) => {
  return <div className='c-ui-multipleSelect-multipleValueRemove'>
    <Icons kind='close'/>
  </div>
}

export default MultipleValueRemove


