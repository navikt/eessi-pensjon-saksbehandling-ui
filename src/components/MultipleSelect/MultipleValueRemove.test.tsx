import { screen, render } from '@testing-library/react'
import MultipleValueRemove from './MultipleValueRemove'

describe('components/MultipleSelect/MultipleValueRemove', () => {
  const initialMockProps = {}

  it('Render: match snapshot', () => {
    const { container } = render(<MultipleValueRemove {...initialMockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    render(<MultipleValueRemove {...initialMockProps} />)
    expect(screen.getByTestId('c-multipleselect-multiplevalueremove')).toBeInTheDocument()
  })
})
