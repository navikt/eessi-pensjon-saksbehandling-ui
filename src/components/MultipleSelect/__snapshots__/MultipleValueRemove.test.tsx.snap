// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MultipleSelect/MultipleValueRemove Render: match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c1:hover {
  background-color: transparent !important;
}

ui-fss.g1[id="c0"] {
  content: "c1,";
}

<MultipleValueRemove>
  <styled.div
    data-testid="c-multipleselect-multiplevalueremove"
    style={Object {}}
  >
    <div
      className="c0 c1"
      data-testid="c-multipleselect-multiplevalueremove"
      style={Object {}}
    >
      <ForwardRef(SvgErrorFilled)
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        <svg
          fill="none"
          focusable={false}
          height="1em"
          role="img"
          style={
            Object {
              "visibility": "visible",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0zm3.571 7L17 8.429 13.428 12 17 15.571 15.571 17 12 13.428 8.429 17 7 15.571 10.572 12 7 8.429 8.429 7 12 10.572 15.571 7z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </ForwardRef(SvgErrorFilled)>
    </div>
  </styled.div>
</MultipleValueRemove>
`;
